### DISCUSSION TITLE
Option not to enforce foreign keys?

### DISCUSSION THREAD

**Discussion by Phil-T1** [duckdb/duckdb#6127] [Ideas]
Hi,

Is it possible to implement a NO CHECK foreign key feature?

I can't see any way to disable foreign keys and would like to have the relationships in the schema for reference, but not enforced.

Or is there already another way to do this that I can't find?

Thanks very much,

Phil.

**Comment by bendnorman** [comment]
This would be a great feature! [MySQL](https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_foreign_key_checks) and [SQLite](https://www.sqlite.org/pragma.html#pragma_foreign_keys) support foreign key disabling. SQLite also offers a PRAGMA called `foreign_key_check ` for checking foreign key constraints.

We utilize these sqlite features in the [Public Utility Data Liberation project (PUDL) ](https://github.com/catalyst-cooperative/pudl)where we do a lot of local data processing using dagster. Being able to disable foreign keys, then check them once the data has been loaded in duckdb would be helpful for ETLs that load tables in an order that doesn't satisfy FK constraints but still provide referential integrity checks. DBT tests offers some [referential integrity checks](https://docs.getdbt.com/reference/resource-properties/data-tests#relationships) that are database agnostic. We don't use dbt and it would be nice to include the FK constraints in the duck db schema.

**Reply by zaneselvans** [reply]
Past related issues / discussions include: #4201 #4204 #4205
